packages/styles/src/sass/components/text-input/_index.scss

```diff
@@ -0,0 +1,233 @@
+    }
+  }
+
+  &:invalid {
+    background-color: var(--cbp-color-danger-lighter);
+    border-color: var(--cbp-color-danger-base);
+
+    &:focus {
+      border-color: var(--cbp-color-interactive-focus-dark);
+    }
+    
+    &:hover {
+      border-color: var(--cbp-color-danger-dark);
+    }
+
+    &:hover:focus {
+      border-color: var(--cbp-color-interactive-focus-dark);
+    }
+  }
+
+  // This removes the default increment arrows in the number input field
+  &::-webkit-outer-spin-button,
+  &::-webkit-inner-spin-button {
+    -webkit-appearance: none;
+    margin: 0; /* Margins still appear even if hidden */
+  }
+}
+
+textarea.cbp-input {
+  min-height: 6.75rem;
+  padding-top: var(--cbp-space-2x);
+  padding-bottom: var(--cbp-space-2x);
+  resize: none;
+}
+
+// Inputs with <button> attached to the end
+.cbp-input-group {
+  display: flex;
+
+  & > .cbp-input {
+    border-right: 0;
+    border-top-right-radius: 0;
+    border-bottom-right-radius: 0;
+    flex: 1 1 auto;
+
+    &:disabled {
+      cursor: pointer;
+    }
+
+    & + button {
+      border-bottom-left-radius: 0;
+      border-top-left-radius: 0;
+      white-space: nowrap;
+    }
+
+    & + button:disabled {
+      border-color: var(--cbp-color-interactive-disabled-dark);
+      background-color: var(--cbp-color-interactive-disabled-dark);
+      color: var(--cbp-color-interactive-disabled-light);
+    }
+  }
+}
+
+.cbp-input__numeric-counter {
+  display: flex;
+  gap: var(--cbp-space-4x);
+  
+  & > .cbp-input {
+    -moz-appearance: textfield; // Removes the default increment arrows in Firefox 
+  }
+}
+
+.cbp-input__numeric-switch {
+  display: flex;
+  gap: var(--cbp-space-4x);
+  
+  & > div:first-child {
+    align-items: center;
+    border-color: var(--cbp-color-interactive-neutral-dark);
+    border-radius: var(--cbp-border-radius-softer);
+    border-style: solid;
+    border-width: var(--cbp-border-size-md);
+    display: inherit;
+    height: 2.25rem;
+    width: 100%;
+
+    &:hover {
+      border-color: var(--cbp-color-interactive-neutral-darker)
+    }
+
+    &:focus-within {
+      border-color: var(--cbp-color-interactive-focus-dark);
+      outline: 0;
+    }
+
+    &:focus-within:has(.cbp-input:invalid) {
+      border-color: var(--cbp-color-interactive-focus-dark);
+    }
+
+    &:has(.cbp-input:disabled) {
+      border-color: var(--cbp-color-interactive-disabled-dark);
+    }
+
+    &:has(.cbp-input:read-only) {
+      background-color: var(--cbp-color-interactive-neutral-lighter);
+      border-color: var(--cbp-color-interactive-neutral-light);
+    }
+
+    &:has(.cbp-input:invalid) {
+      border-color: var(--cbp-color-danger-base);
+    }
+  }
+
+  & > div:first-child > .cbp-input {
+    -moz-appearance: textfield; // Removes the default increment arrows in Firefox
+    border: none;
+    min-height: 100%;
+    padding-top: 0;
+    padding-bottom: 0;
+  }
+
+  & > div:first-child::after {
+    content: 'lbs';
```

Comment:
This doesn't lend itself to updating very well. Also noted in my main comments.

Reply:
we could add a span element and change with JS, let's tackle this once we nail down the functionality of the numeric switch pattern.